src/printf/BodyInstrumentation.cpp

```diff
@@ -0,0 +1,99 @@
+ * the European Commission - subsequent versions of the EUPL (the “Licence”).
+ * You may not use this work except in compliance with the Licence.
+ * You may obtain a copy of the Licence at:
+ *
+ *     http://ec.europa.eu/idabc/eupl.html
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the Licence is distributed on an “AS IS” basis, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  Licence permissions and limitations under the Licence.
+ */
+
+#include "BodyInstrumentation.hpp"
+
+#include <fmt/chrono.h>
+#include <fmt/core.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <chrono>
+#include <numeric>
+#include <vector>
+
+auto enter_body(sector* sec [[clang::use_handle("bactria sector")]], char const* source, std::uint32_t lineno) -> void
+{
+    auto const now = std::chrono::steady_clock::now();
+    auto data_ptr = static_cast<BodyData*>(sec->data.get());
+
+    ++data_ptr->iterations;
+
+    if(!data_ptr->first_is_set)
+    {
+        data_ptr->global.first = now;
+        data_ptr->first_is_set = true;
+    }
+
+    data_ptr->locals.emplace_back(now, now);
+
+    auto const parents = sector_stack.size(); 
+    fmt::print("{:>{}} Entering body {} - {}:{}.\n", "|", 4u * parents, sec->name, source, lineno);
+    sector_stack.push(sec);
+}
+
+auto leave_body(sector* sec [[clang::use_handle("bactria sector")]], char const* source, std::uint32_t lineno) -> void
+{
+    auto const now = std::chrono::steady_clock::now();
+    auto data_ptr = static_cast<BodyData*>(sec->data.get());
+
+    data_ptr->global.second = now;
+    data_ptr->locals.back().second = now;
+
+    auto const parents = sector_stack.size();
+    fmt::print("{:>{}} Leaving body {} - {}:{}.\n", "|", 4u * (parents - 1u), sec->name, source, lineno);
+    handle_pop(sec);
+}
+
+auto body_summary(sector* sec [[maybe_unused, clang::use_handle("bactria sector")]]) -> void
+{
+    auto data_ptr = static_cast<BodyData*>(sec->data.get());
+
+    using precise_duration = std::chrono::duration<double, std::micro>;
+
+    auto dur_vec = std::vector<double>(data_ptr->locals.size());
+    std::transform(std::begin(data_ptr->locals), std::end(data_ptr->locals),
+                   std::begin(dur_vec),
+                   [](TimePair const& p)
+                   {
+                        return (std::chrono::duration_cast<precise_duration>(p.second - p.first)).count(); 
+                   });
+
+    // duration
+    auto const dur = std::chrono::duration_cast<precise_duration>(data_ptr->global.second - data_ptr->global.first);
+
+    // min, max
+    auto const minmax_ticks = std::minmax_element(std::begin(dur_vec), std::end(dur_vec));    
+    auto const min = precise_duration{*minmax_ticks.first};
+    auto const max = precise_duration{*minmax_ticks.second};
+
+    // average
+    auto const sum_ticks = std::accumulate(std::begin(dur_vec), std::end(dur_vec), 0.0);
```

Comment:
```suggestion
    auto const sum_ticks = std::reduce(std::begin(dur_vec), std::end(dur_vec));
```

Reply:
Sadly, `reduce` is not available in C++14. It was introduced with C++17.